### PR TITLE
Update from CICE Consortium main #cb58257857

### PR DESCRIPTION
**Pull request by apcraig**
## PR checklist
- [X] Short (1 sentence) summary of your PR: 
    Update from CICE Consortium main #cb58257857
- [X] Developer(s): 
    apcraig
- [X] Suggest PR reviewers from list in the column to the right.
- [X] Please copy the PR test results link or provide a summary of testing completed below.
    full test suite on cheyenne is bit-for-bit, https://github.com/CICE-Consortium/Test-Results/wiki/cice_by_hash_forks#d14a3a79b2adc359a78e13fd0e13b9767874649f vs Consortium main in log files, not bit-for-bit in restart files due to differences in Tsfcn over non-ice points.
- How much do the PR code changes differ from the unmodified code? 
    - [X] bit for bit
    - [ ] different at roundoff level
    - [ ] more substantial 
- Does this PR create or have dependencies on Icepack or any other models?
    - [ ] Yes
    - [X] No
- Does this PR update the Icepack submodule?  If so, the Icepack submodule must point to a hash on Icepack's main branch.
    - [ ] Yes
    - [X] No, points to current hash on E3SM-Project/Icepack/cice-consortium/E3SM-icepack-initial-integration
- Does this PR add any new test cases?
    - [X] Yes, just changes from Consortium main
    - [ ] No
- Is the documentation being updated? ("Documentation" includes information on the wiki or in the .rst files from doc/source/, which are used to create the online technical docs at https://readthedocs.org/projects/cice-consortium-cice/. A test build of the technical docs will be performed as part of the PR testing.)
    - [X] Yes
    - [ ] No, does the documentation need to be updated at a later time?
        - [ ] Yes
        - [ ] No 
- [X] Please provide any additional information or relevant details below:

Syncs E3SMIcepackDEV with #cb58257857 on Consortium main
Includes synced E3SM-Project/Icepack/cice-consortium/E3SM-icepack-initial-integration #4d131da975b17e8df5 synced with CICE-Consortium/Icepack/main #cabfe0f111b6

### CODE REVIEW

**Comment by apcraig** [comment]
Running full test suite now to recheck, should be bit-for-bit with Consortium main #cb58257857.

**Comment by apcraig** [comment]
full test suite on cheyenne is bit-for-bit, https://github.com/CICE-Consortium/Test-Results/wiki/cice_by_hash_forks#d14a3a79b2adc359a78e13fd0e13b9767874649f vs Consortium main in log files, not bit-for-bit in restart files due to differences in Tsfcn over non-ice points.

File: cicecore/cicedyn/general/ice_step_mod.F90
Comment (eclare108213) [inline cicecore/cicedyn/general/ice_step_mod.F90:120]: what does 'icep' signify?

File: cicecore/cicedyn/analysis/ice_diagnostics.F90
Comment (eclare108213) [inline cicecore/cicedyn/analysis/ice_diagnostics.F90:429]: kinetic energy is still BFB in the log files?

**Review by eclare108213 — APPROVED** [review]
Wow, 180 files with differences!

**Comment by apcraig** [comment]
A lot of the 180 file diffs are renaming from cicedynB to cicedyn and changing permissions on files.

File: cicecore/cicedyn/analysis/ice_diagnostics.F90
Comment (apcraig) [reply cicecore/cicedyn/analysis/ice_diagnostics.F90:429]: I am comparing to the current CICE Consortium main.  So we expect/hope that will be bit-for-bit.  If I compared against the most recent version of E3SM CICE, then I expect the log files would be different.

File: cicecore/cicedyn/general/ice_step_mod.F90
Comment (eclare108213) [reply cicecore/cicedyn/general/ice_step_mod.F90:120]: Oh, it's probably for 'icepack'.  That wasn't intuitive to me.

File: cicecore/cicedyn/analysis/ice_diagnostics.F90
Comment (eclare108213) [reply cicecore/cicedyn/analysis/ice_diagnostics.F90:429]: okay, that makes sense

File: cicecore/cicedyn/general/ice_step_mod.F90
Comment (apcraig) [reply cicecore/cicedyn/general/ice_step_mod.F90:120]: _icep means "passed into icepack".  This is a pretty awkward implementation.  [u,v]velT_icep are the ice velocities passed into icepack for computation of the wind stress.  If highfreq is true, then [u,v]velT_icep = [u,v]vel interpolated to the T grid.  If highfreq is false, then [u,v]velT_icep=0.  The implementation was refactored recently to make this clearer, but it's still awkward.  In my opinion, highfreq and [u,v]velT should probably be passed into icepack and then icepack should decide what to do.  Even better, it shouldn't be called highfreq.  The option should be called something like "compute_wind_stress_relative_to_ice_velocity".  It might mostly be used with highfreq coupling, but I think how this is setup is a bit confusing.

**Comment by eclare108213** [comment]
Once this settles, please, would you summarize the current state of the various repos in terms of what's identical, what's not identical but is BFB, etc?  I'll add that info to the hackathon page for next Monday.  Then I'll start diff'ing code.  Thanks - it's clearly a huge amount of work to keep everything synced up.

**Comment by apcraig** [comment]
I will summarize once settled.  I'm actually looking at one more set of mods to the E3SM Icepack to bring the restart files back to bit-for-bit.  The difference is the tsfcn over ice free ocean.  I think we should just revert that back to the old computation in it's entirety temporarily to keep testing more robust vs Consortium main.  More soon.  But it won't affect this PR.  I'll do it separately.

**Comment by eclare108213** [comment]
Let's think about the options for Tsfc over ice-free ocean before you revert.  That is one of the things that changed in E3SM, which we are bringing into Icepack.  So maybe it's better to change Consortium main now?  Or is there still uncertainty about the implementation?

**Comment by apcraig** [comment]
The problem is that changing Tsfc in the Consortium is that it would require a change to the Icepack interface, passing in Tf.  I don't think we want to do that right now.

I just ran a quick test to see how easy it might be to revert in the E3SM version.  It's more complicated that I hoped.  It would probably involve reverting the Tsfc in a few places.  There seems to be some Tsfc issue with respect to very small ice fractions that is complicating the situation.  I think this is why we need the "_old" option sometimes (but not always).  There may be a small bug there somewhere related to "puny" or similar, but it certainly doesn't impact science.

I think we may have to leave this all as is for the moment.

**Comment by eclare108213** [comment]
Ah, yes, the residual ice problem keeps popping up.  :(    I hope to have more time to work on that after this big push for the Icepack merge.  But of course the BGC is going to take a while.

**Comment by apcraig** [comment]
E3SM Icepack (https://github.com/E3SM-Project/Icepack/tree/cice-consortium/E3SM-icepack-initial-integration) #4d131da, Nov 20, 2022 is bit-for-bit with CICE-Consortium Icepack (https://github.com/CICE-Consortium/Icepack) #cabfe0f, Nov 20, 2022 except for the modal aerosol tests which diverges in the full_ITD case after 15 days.  I'm not sure we had done standalone Icepack modal aerosol tests between the two versions prior to now.

E3SM CICE (https://github.com/apcraig/CICE/tree/E3SMIcepackDEV) #ec8632f, Nov 21, 2022 is bit-for-bit with CICE-Consortium CICE (https://github.com/CICE-Consortium/CICE) #cb58527, Nov 20, 2022.  However, the restart files may be different in Tsfc in ice-free ocean areas.  This does not affect science but does affect ability to do regression testing using the restart files.  Log files are also compared in that case and they indicate results are scientifically identical.  These versions of CICE include the respective Icepack versions above.  One thing to note, the modal aerosol tests in CICE are typically only 5 or 10 days, I don't know if that is hiding the bit-for-bit difference we see above after 15 days.

**Pull request merged**
